resources/assets/js/editor.js

```diff
@@ -148,7 +148,21 @@ function setupEditor(response) {
 }
 function addTab(file) {
     let span = document.createElement('span');
-    span.innerHTML = file.filename;
+    span.dataset.filename = file.filename;
+    let text = document.createTextNode(file.filename);
+    span.appendChild(text);
+    if(file.filename != 'main.py') {
+      let deleteButton = document.createElement('span');
+      deleteButton.innerHTML = '&times;';
+      deleteButton.onclick = function(this) {
```

Comment:
The "this" parameter is causing an error for me. Should this be event?

Reply:
Oops! Don’t know why I ever did that..
Just goes to show that however confident you are in your code, you should always test it!
Thanks; fixed.